examples/kuka_rsi_robot_tests/kuka_rsi_robot_tests/quantec_test.py

```diff
@@ -0,0 +1,310 @@
+        self.get_state()
+
+    def deactivate_done_callback(self, future):
+        self.get_logger().info(
+            "deactivate call done. If it failed, probably manual intervention is required"
+        )
+        self.movement_done.set()
+
+    def activate_robot(self):
+        self.send_configure()  # this will in callback send the activate
+
+    def get_state(self):
+        state_future = self.state_client.call_async(GetState.Request())
+        state_future.add_done_callback(self.get_state_callback)
+        self.get_logger().info("waiting for getting state")
+
+    def get_state_callback(self, future):
+        self.get_logger().info("got current state")
+        state_resp = future.result()
+        if state_resp.current_state.id == State.PRIMARY_STATE_ACTIVE:
+            self.get_logger().info("Robot is active")
+            self.current_index = 0
+            self.start_next_movement()
+        else:
+            self.get_logger().error("could not activate robot")
+            if self.start_service:
+                self.movement_done.set()
+            else:
+                self.executor.shutdown()
+
+    def start_next_movement(self):
+        if self.current_index < len(self.times):
+            self.get_logger().info("going to submit movement")
+            position = self.moves[self.current_index]
+            duration = self.times[self.current_index]
+            self.current_index = self.current_index + 1
+            self.start_robot_movement(position, duration)
+        else:
+            self.get_logger().info("All movements done")
+            if self.start_service:
+                self.movement_done.set()
+            else:
+                self.executor.shutdown()
+
+    def start_robot_movement(self, position, duration):
+        goal = FollowJointTrajectory.Goal()
+        trajectory = JointTrajectory()
+        trajectory.joint_names = ["joint_1", "joint_2", "joint_3", "joint_4", "joint_5", "joint_6"]
+        goal_point = JointTrajectoryPoint()
+        goal_point.time_from_start = Duration(sec=duration)
+        goal_point.positions = position
+        trajectory.points.append(copy.deepcopy(goal_point))
+        goal.trajectory = trajectory
+        self.action_client.wait_for_server()
+        cur_action_submit = self.action_client.send_goal_async(goal)
+        self.get_logger().info("waiting for action to be submitted")
+        # self.executor.spin_until_future_complete(cur_action_submit)
+        cur_action_submit.add_done_callback(self.action_submit_callback)
+
+    def action_submit_callback(self, future):
+        self.get_logger().info("action submitted, waiting for action to be done")
+        result_action = future.result().get_result_async()
+        result_action.add_done_callback(self.action_done_callback)
+
+    def action_done_callback(self, future):
+        self.get_logger().info("action finished in callback")
+        self.start_next_movement()
+
+    def start_robot_movements(self):
```

Comment:
is this used?

Reply:
hmmm. Seems like it is not used anymore. will remove it.